### PR TITLE
Use submodules

### PR DESCRIPTION
**Pull request by danewalton**
This PR currently points to the `dane/feature/iot-adu` branch of the middleware for FreeRTOS.

https://github.com/Azure/azure-iot-middleware-freertos/pull/238

### CODE REVIEW

**Review by ewertons — APPROVED** [review]

**Comment by danewalton** [comment]
I will merge this once all the bottom ones are merged.

**Pull request merged**
